tests/e2e/analyze_graph/optimize_indexes.py

```diff
@@ -0,0 +1,344 @@
+# Copyright 2023 Memgraph Ltd.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+import sys
+
+import pytest
+from common import connect, execute_and_fetch_all
+
+# E2E tests for checking query semantic
```

Comment:
One top level comment. It seems to me there is no test when we have nodes with different labels here. I would say we need such test also, to cover for such case.